scripts/library.sh

```diff
@@ -315,6 +315,51 @@ function check_licenses() {
   run_dep_collector -check $@
 }
 
+# Run gofmt to check format of .go files
+function check_format() {
+  if [[ -z "${GOROOT}" ]]; then
+    local GOROOT=$(go env GOROOT)
+  fi
+  gofmt="${GOROOT}/bin/gofmt"
+
+  diff=$(  find . -not \( \
+      \( \
+        -wholename './output' \
+        -o -wholename './_output' \
+        -o -wholename './_gopath' \
+        -o -wholename './release' \
+        -o -wholename './target' \
+        -o -wholename '*/third_party/*' \
+        -o -wholename '*/vendor/*' \
+      \) -prune \
+    \) -name '*.go' \
+   | xargs ${gofmt} -d -s 2>&1) || true
+  if [[ -n "${diff}" ]]; then
+    echo "${diff}" >&2
+    echo >&2
+    echo "Run ./hack/update-gofmt.sh" >&2
+    exit 1
+  fi
+  echo `date`,"No errors found."
+}
+
+# Run misspell to check the spelling
+function check_spelling() {
+  local misspell="$(which misspell)"
+  if [[ -z ${misspell} ]]; then
+    go get -u github.com/client9/misspell/cmd/misspell
```

Comment:
this can be part of the prow(test-infra/images/prow-tests/) image, before running the tests

Reply:
k8s already has a script https://github.com/kubernetes/test-infra/blob/master/hack/verify-gofmt.sh that does gofmt. We can run that with Prow as well instead of creating a new plugin